internal/printer/printer.go

```diff
@@ -458,6 +483,27 @@ func maybeConvertTransition(n *astro.Node) {
 			Type: astro.ExpressionAttribute,
 		})
 	}
+	if transform.HasAttr(n, transform.TRANSITION_PERSIST) {
+		transitionPersistIndex := transform.AttrIndex(n, transform.TRANSITION_PERSIST)
+		// If there no value, create a transition scope for this element
+		if n.Attr[transitionPersistIndex].Val != "" {
+			// Just rename the attribute
+			n.Attr[transitionPersistIndex].Key = "data-astro-transition-persist"
+
+		} else if transform.HasAttr(n, transform.TRANSITION_NAME) {
```

Comment:
If there is a transition:name on the element, use that as the persist value as well.

`<div transition:name="media-player" transition:persist>` This just avoids repeating the same thing.